src/shared/steps/setup-device-step/SetupDeviceStep.js

```diff
@@ -0,0 +1,205 @@
+    nameSufix: 'Device',
+};
+
+class SetupDeviceStep extends Component {
+    state = {
+        selectedOption: null,
+        activeSubStepIndex: 1,
+    };
+
+    formRef = createRef();
+
+    constructor(props) {
+        super(props);
+
+        this.detectedDeviceInfo = deviceInfo ? deviceInfo : DEVICE_INFO_FALLBACK;
+        this.selectedDeviceInfo = this.getSelectedDeviceInfo(this.detectedDeviceInfo.type);
+        this.formInitialValues = {
+            'identity-device': this.selectedDeviceInfo.id,
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.identityFirstName && this.props.identityFirstName) {
+            this.setTextInputValue(this.props.identityFirstName, this.detectedDeviceInfo.nameSufix);
+        }
+
+        if (prevProps.identityFirstName !== this.props.identityFirstName) {
+            const sufix = !this.state.selectedOption ? this.detectedDeviceInfo.nameSufix : this.selectedDeviceInfo.label;
+            const prevDeviceName = this.getDefaultDeviceName(prevProps.identityFirstName, sufix);
+            const currentDeviceName = this.formRef.current.form.getFieldState('device-name').value;
+
+            if (prevDeviceName === currentDeviceName) {
+                this.setTextInputValue(this.props.identityFirstName, this.selectedDeviceInfo.label);
+            }
+        }
+    }
+
+    componentWillUnmount() {
+        this.substepTransitionTimeout && clearTimeout(this.substepTransitionTimeout);
+    }
+
+    render() {
+        const { stepData: { title, description, buttonText } } = this.props;
+        const { activeSubStepIndex } = this.state;
+        const { label, icon } = this.selectedDeviceInfo;
+
+        return (
+            <div className={ styles.contentWrapper }>
+                <h2 className={ styles.heading }>{ title }</h2>
+                <p>{ description }</p>
+
+                <Form
+                    ref={ this.formRef }
+                    initialValues={ this.formInitialValues }
+                    onSubmit={ this.handleOnSubmit }>
+                    { ({ handleSubmit, invalid }) => (
+                        <form autoComplete="off" onSubmit={ handleSubmit }>
+                            <FormSpy onChange={ this.handleFormChange } />
+                            <FaderContainer activeIndex={ activeSubStepIndex }>
+                                { this.renderTypeGroupField() }
+                                <div className={ styles.deviceInfoWrapper }>
+                                    <TypeOption
+                                        label={ label }
+                                        selected
+                                        badge={ <EditIcon /> }
+                                        onClick={ this.handleTypeOptionClick }>
+                                        { icon }
+                                    </TypeOption>
+                                    <Field
+                                        name="device-name"
```

Comment:
For consistency with the future integration of `idm-wallet`, change the name to `name`

Reply:
I've renamed the components:

`<IdentityType>` -> `<IdentityInfo>`
`IdentityInfo` -> `<IdentityDevice>`

I've also changed the `<Field>` names as you requested.